source/Octopus.Server.Client/OctopusAsyncClient.cs

```diff
@@ -633,9 +633,11 @@ protected virtual async Task<OctopusResponse<TResponseResource>> DispatchRequest
                         return octopusResponse;
                     }
                 }
-                catch (TaskCanceledException)
+                catch (TaskCanceledException exception) when (
+                    exception.CancellationToken != cancellationToken &&
+                    exception.InnerException is TimeoutException)
                 {
-                    throw new TimeoutException($"Timeout getting response from {request.Uri} (client timeout is set to {client.Timeout}).");
+                    throw new TimeoutException($"Timeout getting response from {request.Uri} (client timeout is set to {client.Timeout}).", exception);
```

Comment:
This is a nice improvement. It's a shame the current design locks us into a TimeoutException wrapped in a TaskCanceledException wrapped in a TimeoutException.